src/hooks/useSAMLSignIn.ts

```diff
@@ -1,8 +1,23 @@
 import { useCallback } from "react";
 import Bugsnag from "@bugsnag/js";
+
 import firebase from "firebase/app";
 
-export const useSAMLSignIn = (samlAuthProviderId?: string) => {
+import { ReactHook } from "types/utility";
+
+export interface UseSAMLSignInProps {
+  samlAuthProviderId?: string;
+}
+
+export interface UseSAMLSignInReturn {
+  hasSamlAuthProviderId: boolean;
+  signInWithSAML: () => void;
+}
+
+export const useSAMLSignIn: ReactHook<
+  UseSAMLSignInProps,
+  UseSAMLSignInReturn
+> = ({ samlAuthProviderId }) => {
   const hasSamlAuthProviderId = samlAuthProviderId !== undefined;
 
   const signInWithSAML = useCallback(() => {
```

Comment:
Use one of the `useAsync` variants here as called out elsewhere in previous review thread.

Reply:
Done in https://github.com/sparkletown/sparkle/pull/1667/commits/36f9ee2f4cfc1f5342c9ef7c0843ba5452393d9f

Reply:
I think you'll need to `await` or `return` that `firebase.auth.etc` `Promise` for this to work properly?